### ISSUE TITLE
Slide and text controls too far away from the element they modify

### ISSUE THREAD

**Issue by jancborchardt** [harish-io/Impressionist#3]
Instead of permanently being in the same bar for all the elements, they should be directly at the element. Like with Photoshop or Gimp, rotating an element is possible by clicking it, then an edge becomes a rotation arrow which you can then drag to rotate. This is much easier to operate than the dislocated rotate and skew controls.

Similar with slide depth and slide scale: the resulting value is already displayed below the slide itself, so why not put the control directly there as control _and_ value display. (Aside: slide depth could be more easily identifiable by changing the color brightness of the blue which frames the slide: brighter=higher, darker=deeper. And similarly slide scale should be directly represented in the overview.)

And the same with the text controls: These are useless unless text is selected. So they should not be displayed while useless. And when some text is selected, they should be displayed close to that element, as a little popover in place.

Also see [principles of proximity: grouping](http://en.wikipedia.org/wiki/Principles_of_grouping#Proximity)

Basically: You can get rid of the black controls bar (also the black bar on the bottom, what does it do?)
In presentation view, the remaining element is just the view switcher, which can be placed in the blue top bar.
In orchestration view, the preview and download buttons can also be placed in the blue top bar. They are even similar to the »file« menu.

**Comment by harish-io** [comment]
In context / proximity based menu is something that I want to definitely do - At the moment, I may spend some time to fix some core authoring specific issues and then step over to do the UI re-work. Thanks a lot for the suggestion, this one is definitely high on the v 0.2 wish list.

**Comment by harish-io** [comment]
Changed the text and image property changing controls into a proximity based menu. Click on an object and the menu appears adjacent to it. The drag to rotate / skew will be deferred to the next release. I will open another issue for that,

**Comment by jancborchardt** [comment]
Great! Is that open for trying out somewhere? I checked [alpha 1](http://hsivaram.com/impressionist/alpha1/) and [alpha 2](http://hsivaram.com/impressionist/alpha2/) but it doesn’t show up for me.

**Comment by jancborchardt** [comment]
If you scroll down on Ihttp://css3please.com there’s a small square you can skew, rotate, move and scale. If the buttons there were icons it would be really good in-place controls.

**Comment by harish-io** [comment]
It is up in the alpha2 branch. Thanks for the pointed on css3please. Will take a look at how this can be implemented into impressionist.

**Comment by harish-io** [comment]
Closed. But reverted the proximity based menu to the static toolbar for better usability after fixing the transform controls issue.

**Comment by jancborchardt** [comment]
I’d still say the proximity based menu is way more usable. When you want to adjust rotation and placement for a single slide, you always need to switch from the slide to the controls. And the same is if you want to adjust depth or rotation for multiple slides.

**Comment by harish-io** [comment]
Agree. I am going to keep the proximity based menu for orchestration view, but not for slide authoring. Just run http://hsivaram.com/impressionist/alpha3 to preview the current change. I personally feel this makes a lot more sense in this view. Thanks!
